### DISCUSSION TITLE
posts/zsh-config-oh-my-zsh/

### DISCUSSION THREAD

**Discussion by giscus** [leegical/leegical.github.io#7] [Announcements]
# posts/zsh-config-oh-my-zsh/

传统的 bash 功能比较简陋，且不美观。本文基于 Ubuntu22.04 LTS 系统，安装 zsh，并使用 oh-my-zsh 对终端进行美化。Oh My Zsh 是基于 zsh 命令行的一个扩展工具集，提供了丰富的扩展功能。

https://www.haoyep.com/posts/zsh-config-oh-my-zsh/

**Comment by OldSaltFish** [comment]
文章加个目录应该会好一些

**Reply by leegical** [reply]
一直都有目录的啊

**Reply by popstarmc** [reply]
右侧有目录的

**Comment by wandaren** [comment]
漏了一个细节，新安装了 Oh My Zsh后原来配置在.bashrc中的环境信息会无法生效

**Reply by leegical** [reply]
是的。一般zsh的默认的环境配置文件是`~/.zshrc`，可以手动把`~/.bashrc`里的自定义环境变量复制过来

**Reply by yuyi2439** [reply]
也可以搞一个.shrc，然后在.bashrc和.zshrc中都引用这个新建的.shrc

**Comment by MesmerizingXiaoSenSen** [comment]
谢谢你的文章，很清晰，按照你的流程完全无误，没有各种错误和中断

**Comment by donotknowsjtu** [comment]
感谢博主，非常有用的教程

**Comment by mistcoversmyeyes** [comment]
博主写的配置过程很详细，一遍过成功配置完成了。

**Comment by Ashinch** [comment]
很好的 guideline 👍

**Comment by lizihao-devx** [comment]
赞！很有用
跟着指南一遍就配置完成了，还有我很苦恼的终端代理也解决了
感谢博主

**Comment by NathanYYDS** [comment]
好文章！！！

**Comment by kanwofeiwo** [comment]
作者大大，请问设置为默认时报错chsh: /bin/zsh is an invalid shell
是为什么啊😦

**Reply by HFrancisla** [reply]
我不是作者，我这是看到了你的问题，你可以尝试使用简单的find指令
`find /usr -name zsh  `
来尝试寻找zsh，这可能能帮助到你，祝你好运

**Reply by leegical** [reply]
可能
1. 没有安装zsh，手动安装一下再尝试
2. 你所使用的系统，zsh二进制不在`/bin/zsh`，请自行搜索一下

**Comment by kanwofeiwo** [comment]
谢谢你！解决了，感谢感谢🙏🏻



---Original---
From: ***@***.***&gt;
Date: Mon, Dec 9, 2024 11:43 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [leegical/leegical.github.io] posts/zsh-config-oh-my-zsh/(Discussion #7)




 
可能
  
没有安装zsh，手动安装一下再尝试
 
你所使用的系统，zsh二进制不在/bin/zsh，请自行搜索一下

**Comment by lwhluvdemo** [comment]
非常有用,感谢.

**Comment by kanwofeiwo** [comment]
谢谢你！



---Original---
From: ***@***.***&gt;
Date: Sat, Dec 7, 2024 22:07 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [leegical/leegical.github.io] posts/zsh-config-oh-my-zsh/(Discussion #7)




 
我不是作者，我这是看到了你的问题，你可以尝试使用简单的find指令
 find /usr -name zsh  
 来尝试寻找zsh，这可能能帮助到你，祝你好运

**Comment by kanwofeiwo** [comment]
谢谢



---Original---
From: ***@***.***&gt;
Date: Mon, Dec 9, 2024 11:43 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [leegical/leegical.github.io] posts/zsh-config-oh-my-zsh/(Discussion #7)




 
可能
  
没有安装zsh，手动安装一下再尝试
 
你所使用的系统，zsh二进制不在/bin/zsh，请自行搜索一下

**Comment by Savagel0ve** [comment]
很好的文章 以前需要找很多文章 东拼西凑 这次终于可以一条龙配完

**Comment by dazhaoDai** [comment]
很准确，很详细

**Comment by OldSaltFish** [comment]
想问问你对fish shell的看法

**Reply by leegical** [reply]
对bash脚本兼容性不好，不太适合用在生产环境

**Comment by Lezy233** [comment]
好文章，非常适合我这种小白

**Comment by yyds1w3** [comment]
好耶
谢谢博主指导

**Comment by dongm2ez** [comment]
文章不错

**Comment by JiaqiJiang1206** [comment]
感谢！很有帮助！

**Comment by yangyi0570** [comment]
good article, love from China :)

**Comment by re-imagined** [comment]
最完备的最贴心的 zsh 配置教程 👍

**Comment by qlAD** [comment]
这个终端字体是什么 ：）

**Comment by xins981** [comment]
使用第三方主题 pure 时，千万不要填写 zsh_theme 字段。
否则会报 [oh-my-zsh] theme 'pure' not found

正确姿势如下：
vim .zshrc

追加以下三行：
fpath+=($HOME/.zsh/pure)
autoload -U promptinit; promptinit
prompt pure

source .zshrc

**Comment by yscheng04** [comment]
谢谢，写的很好
